File: python/1-9/problem3.py
from math import sqrt, ceil
from lib.primes import primes_sieve

prime_factors = []
for prime in primes_sieve(int(ceil(sqrt(600851475143)))):
    if 600851475143 % prime == 0:
        prime_factors.append(prime)

print(prime_factors)

File: ruby/1-9/problem3.rb
# The prime factors of 13195 are 5, 7, 13 and 29.

# What is the largest prime factor of the number 600851475143?

require_relative "../lib/primes"

def prime_factors(n)
  prime_limit = Math.sqrt(n).ceil
  Primes.sieve(prime_limit).select do |prime|
    n % prime == 0
  end
end


puts prime_factors(600_851_475_143)[-1]
